### PR TITLE
feat: [BB-189] Feature** implemented back button

### PR DESCRIPTION
**Pull request by kcaparas1630**
# Description

Created a Back Button so there is a user-friendly option to go back to the form instead of only using the form header as a way to go back and go forward to the form.

> Given that I press back on the Field and Soil Form Module it will redirect me to Farm Information Module

from:
image
to:
image

Note that this pull request doesn't include storing user inputs into local storage. So whenever going back to field and soil form module it will display the input fields first then will still show the earlier list that was inputted. It will be implemented on the future sprint.

## Type of changes

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-203-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-203-backend.apps.silver.devops.gov.bc.ca/api/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-203-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-203-backend.apps.silver.devops.gov.bc.ca/api/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/Views/MainPage/MainPage.tsx [hunk 1]
```diff
@@ -62,15 +62,31 @@ const MainPage: React.FC = () => {
...
+        return;
+    }
+    // if (moveBack) {
+    //   secondModuleID = formStates[currForm - 1].id;
+    //   setCurrForm((prevForm) => prevForm - 1);
+    // }
+    // if (moveForward) {
+    //   secondModuleID = formStates[currForm + 1].id;
+    //   setCurrForm((prevForm) => prevForm + 1);
+    // }
```
Comment (GDamaso) [inline frontend/src/Views/MainPage/MainPage.tsx]: We forgot to remove this

File: frontend/src/Views/MainPage/MainPage.tsx [hunk 2]
```diff
@@ -91,19 +107,12 @@ const MainPage: React.FC = () => {
...
-        return prevForm - 1;
-      }
-      return prevForm;
-    });
-  };
-
+  // const handleBackState = () => {
+  //   handleFormState(formStates[currForm].id, true);
+  //   setCurrForm((prevForm: number): number => prevForm - 1);
+  // };
```
Comment (GDamaso) [inline frontend/src/Views/MainPage/MainPage.tsx]: And this

File: frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsButtonComponent.tsx [hunk 3]
```diff
@@ -44,7 +45,7 @@ const FieldsButtonComponent: React.FC<ButtonComponentProps> = ({
           disabled={false}
           actions="secondary"
           text={ComponentText.BACK}
-          handleClick={handleBackState}
+          handleClick={() => handleFormState('back')}
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsButtonComponent.tsx:48]: Should we move this to a clickwrapper?

**Review by GDamaso — APPROVED** [review]
Looks good, we forgot to remove a couple commented out code.

File: frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsButtonComponent.tsx (hunk 3 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsButtonComponent.tsx:48]: yeah let's move this to a clickwrapper.

File: frontend/src/Views/MainPage/MainPage.tsx (hunk 1 above)
Comment (kcaparas1630) [reply frontend/src/Views/MainPage/MainPage.tsx]: This is outdated, we cleaned it yesterday. 😛

File: frontend/src/Views/MainPage/MainPage.tsx (hunk 2 above)
Comment (kcaparas1630) [reply frontend/src/Views/MainPage/MainPage.tsx]: also this 😛

**Pull request merged**